Source/SwiftLintFramework/Rules/WeakDelegateRule.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  WeakDelegate.swift
+//  SwiftLint
+//
+//  Created by Olivier Halligon on 11/08/2016.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct WeakDelegateRule: ASTRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.Warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "weak_delegate",
+        name: "Weak Delegate",
+        description: "Delegates should be weak to avoid reference cycles.",
+        nonTriggeringExamples: [
+            "class Foo {\n  weak var delegate: SomeProtocol?\n}\n",
+            "class Foo {\n  weak var someDelegate: SomeDelegateProtocol?\n}\n",
+            "class Foo {\n  weak var delegateScroll: ScrollDelegate?\n}\n",
+            // We only consider properties to be a delegate if it has "delegate" in its name
+            "class Foo {\n  var scrollHandler: ScrollDelegate?\n}\n",
+            // Only trigger on instance variables, not local variables
+            "func foo() {\n  var delegate: SomeDelegate\n}\n"
+        ],
+        triggeringExamples: [
+            "class Foo {\n  var delegate: SomeProtocol?\n}\n",
+            "class Foo {\n  var scrollDelegate: ScrollDelegate?\n}\n",
+            "class Foo {\n  var delegateScroll: ScrollDelegate?\n}\n",
+        ]
+    )
+
+    public func validateFile(file: File,
+                             kind: SwiftDeclarationKind,
+                             dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard kind == .VarInstance else {
+            return []
+        }
+
+        // Check if name contains "delegate"
+        let name = (dictionary["key.name"] as? String) ?? ""
+        let isDelegate = name.lowercaseString.rangeOfString("delegate") != nil
+        guard isDelegate else { return [] }
+
+        // Check if non-weak
+        let attributes = (dictionary["key.attributes"] as? [SourceKitRepresentable])?
```

Comment:
Mmmm wait after re-reading your comment I realize you actually do exclude "weak" in a separate line of the code, that's what I missed and why I didn't understand why you didn't have negative-look-behind in your RegEx.

But still even if we switch to a RegEx rule, I'm not sure I see the advantage of migrating to a RegEx over checking the AST:

* as the AST seems more sure and explicit to me (less likely we forget to match a specific syntax that would represent the same thing, or that our RegEx would be too greedy and catch the "delegate" word in another part of the expression
* For example your regex wouldn't warn on `var delegate: SomeProtocol` if the developer forgot both the `weak` and the optionality of the type… while it should indeed warn for this case too
* I'm not sure a RegEx match using SwiftLint would be able to take the context into account and only match if it's part of a instance variable declaration only? Like, wouldn't your code wrongly match (and warn for) `func foo() { var delegate: SomeDelegate? }` too?
* For example your RegEx would also match and warn for `var scrollHandler: ScrollHandler // maybe we should use delegate: ScrollDelegate? instead?`
* And another example is that your code would consider the lines `var scrollDelegate: ScrollDelegate? // maybe should be weak?` and `var weakLinkDelegate: WeakLinkDelegate?` as valid because they contain "weak" despite not being actually weak.

For all these reasons I prefer the AST analysis, to be sure we consider the right parts of the expression (the name for "delegate" and the attributes for "weak") and the right context "only instance variable")